agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1244,24 +1245,53 @@ func (dg *dockerGoClient) Stats(id string, ctx context.Context) (<-chan *docker.
 		return nil, err
 	}
 
-	stats := make(chan *docker.Stats)
-	options := docker.StatsOptions{
-		ID:                id,
-		Stats:             stats,
-		Stream:            true,
-		Context:           ctx,
-		InactivityTimeout: StatsInactivityTimeout,
-	}
+	returnStats := make(chan *docker.Stats)
 
-	go func() {
-		statsErr := client.Stats(options)
-		if statsErr != nil {
-			seelog.Infof("DockerGoClient: Unable to retrieve stats for container %s: %v",
-				id, statsErr)
+	if !dg.config.PollMetrics {
+		options := docker.StatsOptions{
```

Comment:
Can you add a log line to indicate the behavior, same for the `else` case?

Reply:
Updated.

I added a log for 
"DockerGoClient: Starting to Poll for metrics"
or
"DockerGoClient: Starting to Stream for metrics"

depending on the situation.